cpp/arcticdb/storage/test/test_azure_storage.cpp

```diff
@@ -0,0 +1,121 @@
+/* Copyright 2023 Man Group Operations Limited
+ *
+ * Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+ *
+ * As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+ */
+
+#include <gtest/gtest.h>
+#include <arcticdb/util/test/gtest_utils.hpp>
+
+#include <arcticdb/storage/storage.hpp>
+#include <arcticdb/storage/azure/azure_storage.hpp>
+#include <arcticdb/storage/azure/azure_mock_client.hpp>
+#include <arcticdb/util/buffer.hpp>
+#include <arcticdb/stream/test/stream_test_common.hpp>
+
+using namespace arcticdb;
+using namespace storage;
+using namespace azure;
+
+arcticdb::proto::azure_storage::Config get_mock_azure_config() {
+    arcticdb::proto::azure_storage::Config cfg;
+    cfg.set_use_mock_storage_for_testing(true);
+
+    return cfg;
+}
+
+class AzureMockStorageFixture : public testing::Test {
+protected:
+    AzureStorage store;
+    AzureMockStorageFixture() : store(LibraryPath("_arctic_cfg", '.'), OpenMode::DELETE, get_mock_azure_config()) {}
+};
+
+VariantKey get_variant_key(std::string name) {
```

Comment:
From here down to `list_in_store` is duplicated with `test_s3_storage` except for AzureStorage vs S3Storage. Template that out and re-use?

Reply:
Done in [ab07c4a](https://github.com/man-group/ArcticDB/pull/1331/commits/ab07c4a088ffabddde3a0c43103586332957ab1f)